File: snippets/csharp/batch-messages_main.cs
// using Microsoft.Agents.Xml.Generated.Models;
// using Microsoft.Agents.Protocol.Xml;
// using System.Linq;

// Create batch of messages
var messages = new List<ChatMessage>
{
    new UserMessage { Contents = new List<AIContent> { new TextContent { Text = "Message 1" } } },
    new UserMessage { Contents = new List<AIContent> { new TextContent { Text = "Message 2" } } },
    new UserMessage { Contents = new List<AIContent> { new TextContent { Text = "Message 3" } } }
};

// Process batch
var serializer = new MessageSerializer();
var xmlOutputs = messages.Select(msg => serializer.Serialize(msg)).ToList();

Console.WriteLine($"Processed {xmlOutputs.Count} messages");

File: snippets/python/batch-messages_main.py
# from microsoft.agents.xml.models import ChatMessage, TextContent
# from microsoft.agents.xml.serialization import MessageSerializer
#
# # Create batch of messages
# messages = [
#     ChatMessage(role="user", contents=[TextContent(text="Message 1")]),
#     ChatMessage(role="user", contents=[TextContent(text="Message 2")]),
#     ChatMessage(role="user", contents=[TextContent(text="Message 3")])
# ]
#
# # Process batch
# serializer = MessageSerializer()
# xml_outputs = [serializer.serialize(msg) for msg in messages]
#
# print(f"Processed {len(xml_outputs)} messages")